cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/ProgramRunDispatcherFactory.java

```diff
@@ -42,14 +40,15 @@ public ProgramRunDispatcherFactory(CConfiguration cConf,
    * For unit tests, RemoteProgramRunDispatcher would not be set.
    */
   @Inject(optional = true)
-  public void setRemoteProgramRunDispatcher(@Named(AppFabric.FACTORY_IMPLEMENTATION_REMOTE)
-                                              ProgramRunDispatcher remoteProgramRunDispatcher) {
+  public void setRemoteProgramRunDispatcher(RemoteProgramRunDispatcher remoteProgramRunDispatcher) {
     this.remoteProgramRunDispatcher = remoteProgramRunDispatcher;
   }
 
   public ProgramRunDispatcher getProgramRunDispatcher(ProgramType type) {
+    // TODO(Aditya): Whitelist for ProgramType that can be dispatched remotely needs to be read through configuration.
```

Comment:
to ensure this doesn't get dropped, should open a jira and reference the jira number in the TODO

Reply:
Done. Jira ticket: CDAP-18962